src/sentry/models/grouphash.py

```diff
@@ -38,36 +38,22 @@ class Meta:
         unique_together = (('project', 'hash'), )
 
     @staticmethod
-    def fetch_last_processed_event_id(project_id, group_hash_ids):
-        prefix = 'last-processed-event:{}'.format(project_id)
+    def fetch_last_processed_event_id(group_id, group_hash_ids):
         with redis.clusters.get('default').map() as client:
-            results = map(
-                lambda group_hash_id: client.hget(
-                    '{}:{}'.format(prefix, group_hash_id % 16),
-                    group_hash_id,
-                ),
-                group_hash_ids,
+            result = client.hmget(
+                'gh:lp:{}'.format(group_id),
+                map('{}'.format, group_hash_ids),
             )
 
-        return map(
-            lambda result: result.value,
-            results,
-        )
+        return result.value
 
     @staticmethod
-    def record_last_processed_event_id(project_id, group_hash_ids, event_id):
-        prefix = 'last-processed-event:{}'.format(project_id)
+    def record_last_processed_event_id(group_id, group_hash_id, event_id):
         with redis.clusters.get('default').map() as client:
-            results = map(
-                lambda group_hash_id: client.hset(
-                    '{}:{}'.format(prefix, group_hash_id % 16),
-                    group_hash_id,
-                    event_id,
-                ),
-                group_hash_ids,
+            key = 'gh:lp:{}'.format(group_id)
+            client.hset(
+                key,
+                '{}'.format(group_hash_id),
+                '{}'.format(event_id),
             )
-
-        return map(
-            lambda result: result.value,
-            results,
-        )
+            client.expire(key, 7776000)  # 90d
```

Comment:
Magic number alert — I'm not actually sure if this is a constant value or a setting somewhere…?